From cb4b1b8a0b89dedc7ac0273f4d86bfc793e96832 Mon Sep 17 00:00:00 2001
From: Xulin Sun <xulin.sun@windriver.com>
Date: Thu, 12 Jul 2018 14:43:03 +0800
Subject: [PATCH] drvier: crypto: caam: fsl_qbman/dce replace cpus_allowed to
 cpus_mask for RT

Based on the commit b0f7d3d946(kernel: sched: Provide a pointer to the valid CPU mask)
need to replace cpus_allowed to cpus_mask for RT.

Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/crypto/caam/qi.c |    8 ++++++++
 drivers/dma/dmatest.c    |    4 ++--
 2 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/drivers/crypto/caam/qi.c b/drivers/crypto/caam/qi.c
index f52c684..4dec1e4 100644
--- a/drivers/crypto/caam/qi.c
+++ b/drivers/crypto/caam/qi.c
@@ -496,7 +496,11 @@ int caam_qi_shutdown(struct device *qidev)
 	int i, ret;
 	struct caam_qi_priv *priv = dev_get_drvdata(qidev);
 	const cpumask_t *cpus = qman_affine_cpus();
+#ifdef CONFIG_PREEMPT_RT_FULL
+	struct cpumask old_cpumask = current->cpus_mask;
+#else
 	struct cpumask old_cpumask = current->cpus_allowed;
+#endif
 
 	for_each_cpu(i, cpus) {
 		struct napi_struct *irqtask;
@@ -716,7 +720,11 @@ int caam_qi_init(struct platform_device *caam_pdev)
 	struct device *ctrldev = &caam_pdev->dev, *qidev;
 	struct caam_drv_private *ctrlpriv;
 	const cpumask_t *cpus = qman_affine_cpus();
+#ifdef CONFIG_PREEMPT_RT_FULL
+	struct cpumask old_cpumask = current->cpus_mask;
+#else
 	struct cpumask old_cpumask = current->cpus_allowed;
+#endif
 	static struct platform_device_info qi_pdev_info = {
 		.name = "caam_qi",
 		.id = PLATFORM_DEVID_NONE
diff --git a/drivers/dma/dmatest.c b/drivers/dma/dmatest.c
index 2b47822..5576fff 100644
--- a/drivers/dma/dmatest.c
+++ b/drivers/dma/dmatest.c
@@ -57,10 +57,10 @@
 MODULE_PARM_DESC(sg_buffers,
 		"Number of scatter gather buffers (default: 1)");
 
-static unsigned int dmatest;
+static unsigned int dmatest = 1;
 module_param(dmatest, uint, S_IRUGO | S_IWUSR);
 MODULE_PARM_DESC(dmatest,
-		"dmatest 0-memcpy 1-slave_sg (default: 0)");
+		"dmatest 0-memcpy 1-slave_sg (default: 1)");
 
 static unsigned int xor_sources = 3;
 module_param(xor_sources, uint, S_IRUGO | S_IWUSR);
-- 
1.7.9.5

